tests/unit/server/data-access/MwBotWikibaseMessagesRepo.spec.ts

```diff
@@ -0,0 +1,160 @@
+import TechnicalProblem from '@/common/data-access/error/TechnicalProblem';
+import MessagesNotFound from '@/common/data-access/error/MessageNotFound';
+import MwBotWikibaseMessagesRepo from '@/server/data-access/MwBotWikibaseMessagesRepo';
+import mwbot from 'mwbot';
+import MessageTranslationCollection from '@/datamodel/MessageTranslationCollection';
+import { MessageKeys } from '@/common/MessageKeys';
+
+function newMwBotWikibaseMessagesRepo( bot: any, messageKeys: string[] ) {
+	return new MwBotWikibaseMessagesRepo(
+		bot,
+		messageKeys,
+	);
+}
+
+describe( 'MwBotWikibaseMessagesRepo', () => {
+
+	it( 'can be constructed with mwbot', () => {
+		expect( newMwBotWikibaseMessagesRepo( new mwbot( {} ), Object.values( MessageKeys ) ) )
+			.toBeInstanceOf( MwBotWikibaseMessagesRepo );
+	} );
+
+	describe( 'getMessagesInLanguage', () => {
+		it( 'creates a well-formed allmessages query with amlang', ( done ) => {
+			const inLanguage = 'de';
+			const bot = {
+				request: ( params: object ) => {
+					expect( params ).toEqual( {
+						action: 'query',
+						meta: 'allmessages',
+						ammessages: Object.values( MessageKeys ).join( '|' ),
```

Comment:
This is not incorrect but weird. A variable local to this test containing the options and it being used both here and injected into the constructor (line 38) would make this much more self-explanatory. Think of the test as a documentation of what the code does - "variable injected here, has an effect there".

Reply:
done